### ISSUE TITLE
Endless "Socket is closed" exception

### ISSUE THREAD

**Issue by cora32** [QuickBlox/quickblox-android-sdk#450]
Hello.
An issue when trying to logout.
My logout code:

```
@Override
    public void onSessionClosed(QBRTCSession qbrtcSession) {
        Log.e("QBSessionEventsCallback", "onSessionClosed " + qbrtcSession);
        hangUp(qbrtcSession);

        if (QBRTCClient.getInstance(this) != null) {
            QBRTCClient.getInstance(this).destroy();
        }

        if(QBChatService.getInstance() != null) {
            QBChatService.getInstance().logout(new QBEntityCallback<Void>() {
                @Override
                public void onSuccess(Void aVoid, Bundle bundle) {
                    QBChatService.getInstance().destroy();
                    Log.e(TAG, "Logged out successfuly");
                }

                @Override
                public void onError(QBResponseException e) {
                    e.printStackTrace();
                    Log.e(TAG, "Logging out failed " + e.getLocalizedMessage());
                }
            });
        }
    }
```

results in spamming exception continuously:

```
09-22 14:27:19.533 20540 20644 W System.err: java.net.SocketException: Socket is closed
09-22 14:27:19.533 20540 20644 W System.err: 	at com.android.org.conscrypt.OpenSSLSocketImpl.checkOpen(OpenSSLSocketImpl.java:270)
09-22 14:27:19.533 20540 20644 W System.err: 	at com.android.org.conscrypt.OpenSSLSocketImpl.-wrap0(OpenSSLSocketImpl.java)
09-22 14:27:19.533 20540 20644 W System.err: 	at com.android.org.conscrypt.OpenSSLSocketImpl$SSLInputStream.read(OpenSSLSocketImpl.java:743)
09-22 14:27:19.533 20540 20644 W System.err: 	at java.io.InputStreamReader.read(InputStreamReader.java:184)
09-22 14:27:19.534 20540 20644 W System.err: 	at org.kxml2.io.KXmlParser.peekType(KXmlParser.java:1007)
09-22 14:27:19.534 20540 20644 W System.err: 	at org.jivesoftware.smack.util.PacketParserUtils.parseExtensionElement(Unknown Source)
09-22 14:27:19.534 20540 20644 W System.err: 	at org.jivesoftware.smack.util.PacketParserUtils.addExtensionElement(Unknown Source)
09-22 14:27:19.534 20540 20644 W System.err: 	at org.jivesoftware.smack.util.PacketParserUtils.parseMessage(Unknown Source)
```

Full log:
https://gist.githubusercontent.com/cora32/e850b98a283a24929e4fde33715230ac/raw/2e4855b5f232c690c14d2b4f5ed6bf9c9573ea63/gistfile1.txt

**Comment by tatanka987** [comment]
Don't need call method QBChatService.getInstance().logout  just call QBChatService.getInstance().destroy() and in this method will be called method 'logout' automatically

**Comment by cora32** [comment]
tatanka987, thank you for response. Calling just `QBChatService.getInstance().destroy() ` does not produce the error. 
But now  `onReceiveNewSession()` triggers only once - first time when `<signalType>call</signalType>` arrives and client accepts call. After `onSessionClosed()` all new `<signalType>call</signalType>` won't trigger `onReceiveNewSession()`. 
The reason why i called `logout()` manually is that after this call every new `<signalType>call</signalType>` triggered `onReceiveNewSession()` as they should.

I Also created a question about the problem: https://stackoverflow.com/questions/46365491/why-quickbloxs-onreceivenewsession-called-only-once

**Comment by QB-nikolay-zolotarov** [comment]
Hello QuickBlox customer,

This is Nikolay from QuickBlox support.

The issue was closed as it is outdated.

Please check the relevant section of our documentation here:
[https://docs.quickblox.com/docs/android-authentication](https://docs.quickblox.com/docs/android-authentication)

Also, please update the SDK to the latest version:
[https://github.com/QuickBlox/quickblox-android-sdk-releases/releases/tag/3.9.11](https://github.com/QuickBlox/quickblox-android-sdk-releases/releases/tag/3.9.11)

Additionally, please check our new samples:
[https://docs.quickblox.com/docs/code-samples#chat-samples](https://docs.quickblox.com/docs/code-samples#chat-samples)

If it is still relevant after reviewing the updated information, feel free to open a new issue.

Have a nice day.
